packages/type-utils/tests/getTypeName.test.ts

```diff
@@ -0,0 +1,67 @@
+import { parseForESLint } from '@typescript-eslint/parser';
+import type { TSESTree } from '@typescript-eslint/typescript-estree';
+import path from 'path';
+import type * as ts from 'typescript';
+
+import { getTypeName } from '../src';
+import { expectToHaveParserServices } from './test-utils/expectToHaveParserServices';
+
+describe('getTypeName', () => {
+  function getTypes(code: string): { checker: ts.TypeChecker; type: ts.Type } {
+    const rootDir = path.join(__dirname, 'fixtures');
+
+    const { ast, services } = parseForESLint(code, {
+      project: './tsconfig.json',
+      filePath: path.join(rootDir, 'file.ts'),
+      tsconfigRootDir: rootDir,
+    });
+    expectToHaveParserServices(services);
+    const checker = services.program.getTypeChecker();
+    const declaration = ast.body[0] as TSESTree.TSTypeAliasDeclaration;
+
+    return { checker, type: services.getTypeAtLocation(declaration.id) };
+  }
+
+  function runTest(code: string, expected: string): void {
+    const { type, checker } = getTypes(code);
+    const result = getTypeName(checker, type);
+    expect(result).toBe(expected);
+  }
+
+  describe('returns primitive type', () => {
+    it.each([
+      // string
+      ['type Test = string;', 'string'],
+      ['type Test = "text";', 'string'],
+      ['type Test = string | "text";', 'string'],
+      ['type Test = "string" | "text";', 'string'],
+      ['type Test = string & { foo: number };', 'string'],
+      ['type Test<T = number> = T & string;', 'string'],
+      ['type Test<T extends string = "text"> = T;', 'string'],
+      // number
```

Comment:
Nit: I don't think these comments are super useful. It's pretty apparent what each test is. Most of the sections only have one.

```suggestion
```

Reply:
Thanks! Deleted in acc256f.